mpas_analysis/analysis_task_template.py

```diff
@@ -0,0 +1,227 @@
+    # maybe some tags describing the task.
+    # Keep self and config arguments.
+    # self is a way to access the object that the class MyTask is describing,
+    #    letting you access member variables and methods (member functions),
+    #    as you will see in the examples below.  All methods in a python class
+    #    start with the argument self, which is not included in the list of
+    #    arguments when you call a method of an object (because it is always
+    #    included automatically).
+    # config is an MpasAnalysisConfigParser object that can be used to get
+    #    configuration options stored in config.default or a custom config
+    #    file specific to a given simulation.  See examples below or in
+    #    existing analysis tasks.
+    # Either modify myArg to be a useful additional argument (e.g. the name of
+    # a field to be analyzed) or remove it.
+    def __init__(self, config, myArg='myDefaultValue'):  # {{{
+        '''
+        Construct the analysis task.
+        <Add any additional description of what happens during construction>
+
+        Parameters
+        ----------
+        config :  instance of MpasAnalysisConfigParser
+            Contains configuration options
+
+        myArg : str, optional
+            <Describe the arg, or just remove remove it>
+
+        Authors
+        -------
+        <List of authors>
+
+        '''
+        # first, call the constructor from the base class (AnalysisTask).
+        # Modify MyTask, "myTask", "component" and ['tag1', 'tag2'] below:
+        # taskName is the same as the class name but with a lowercase letter
+        # componentName is one of 'ocean' or 'seaIce' (same as the name of
+        #     the folder where the task resides)
+        # tags are some useful names describing the task ('timeSeries',
+        #     'climatology', 'horizontalMap', 'index', 'transect', etc.) that
+        #     can be used in the 'generate' config option, e.g.
+        #     'all_climatology' of 'no_transect'.  Tasks that include computing
+        # time series, indices and/or climatologies should include
+        # 'timeSeries', 'index' and/or 'climatology' tags, as these are helpful
+        # used to compute and update the start and end dates of each of these
+        # analysis tasks as part of setting up the task.
+        super(MyTask, self).__init__(config=config,
+                                     taskName='myTask',
+                                     componentName='component',
+                                     tags=['tag1', 'tag2'])
+
+        # then, store any additional arguments for use later on.  These would
+        # likely include things like the name of a field, region, month,
+        # season, etc. to be analyzed so that the same subclass of AnalysisTask
+        # can perform several different tasks (potentially in parallel)
+        # Modify or remove the folowing:
```

Comment:
So I don't really know how to address this better than what I already have done here.  If you had a task that was capable of analyzing multiple fields, the field name should be an argument.  I would advocate breaking existing tasks that analyze different seasons into separate tasks, one for each season (or at least having the option to do this), in which case the name of the season ('DJF', 'ANN') would be an argument you would use to construct the task.  But there's no way to know what arguments a task might need in advance.  This is simply part of programming a new task.  I'm just trying to show how you would add arguments containing information you *knew* you needed to pass to the task and couldn't get from config options directly.  We're not currently using this capability, but an example would have been the field names ('sst', 'sss' and 'mld') passed to ocean_modelvsobs.